stan/math/prim/err/check_flag_sundials.hpp

```diff
@@ -1,12 +1,178 @@
 #ifndef STAN_MATH_PRIM_ERR_CHECK_FLAG_SUNDIALS_HPP
 #define STAN_MATH_PRIM_ERR_CHECK_FLAG_SUNDIALS_HPP
 
+#include <kinsol/kinsol.h>
+#include <cvodes/cvodes.h>
 #include <stan/math/prim/meta.hpp>
-#include <stan/math/prim/err/throw_domain_error.hpp>
+#include <stan/math/prim/err/domain_error.hpp>
```

Comment:
Please reorder this to:

#include <stan/math/prim/meta.hpp>
#include <stan/math/prim/err/throw_domain_error.hpp>
#include <stan/math/prim/err/domain_error.hpp>
#include <kinsol/kinsol.h>
#include <cvodes/cvodes.h>


The order we generally try to follow is: Stan Math master header file(like meta.hpp), individual Stan Math header files, std:: header files, other C++ header files, other C header files.

This isn't always possible, so not a strict thing we automatically check,